### ISSUE TITLE
[A11y] Double-labeling fix

### ISSUE THREAD

**Issue by jechols** [OregonDigital/OD2#2010]
### Operating System

Windows

### Browser

Firefox

### Affected page

Collection Show

### Admin user?

Yes

### Issue Description

PR #1981 pulls in a bunch of hyrax code so we can do some overrides, but one oddity gets pulled in that we want to address: in `app/views/hyrax/dashboard/collections/_default_group.html.erb`, there's a `<label for="...">` for screen-reader-only use, then a `<label>` that wraps the same checkbox the `<label for="...">` grabs.

I think the right way to do this kind of thing would be to keep the normal label sighted users have, and then make the hidden text be a description via aria-describedby. The current setup causes Firefox to concatenate the labels as if they were a single label, which can wreck pronunciation (the concatenation doesn't add any spacing) and isn't ideal. The label that's visible is (hopefully) the more important piece, and therefore should be described first I think.

### Can anybody QA or does this need an expert?

A11y familiarity recommended

### QA Steps

This is way easier to test with a screen reader, but you can also inspect the element using firefox's "Inspect Accessibility Properties" item in the context (right-click) menu.
